interop/xds/server/server.go

```diff
@@ -78,8 +87,48 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID}
+
+	for _, val := range getRPCBehaviorMetadata(ctx) {
+		if st := getStatusForRPCBehaviorMetadata(val); st != nil {
+			return response, status.Err(st.Code(), st.Message())
+		}
+	}
+
+	response.Hostname = s.hostname
 	grpc.SetHeader(ctx, metadata.Pairs("hostname", s.hostname))
-	return &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}, nil
+	return response, status.Err(codes.OK, "")
+}
+
+func getRPCBehaviorMetadata(ctx context.Context) []string {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		logger.Error("failed to receive metadata")
+		return nil
+	}
+
+	mdRPCBehavior := md.Get(rpcBehaviorMetadataKey)
+
```

Comment:
Nit: nix this newline please.

Reply:
I generally tend to add a new line wherever the reader might benefit from a logical pause while going down the code

could you help me understand the style directive in play here?